system/jlib/jutil.cpp

```diff
@@ -1617,6 +1617,20 @@ unsigned usTick()
         return 0;
     return (unsigned) ((v.QuadPart*1000000)/freq); // hope dividend doesn't overflow though it might
 }
+extern jlib_decl unsigned __int64 nsTick()
+{
+    static __int64 freq=0;
+    LARGE_INTEGER v;
+    if (!freq) {
+        if (QueryPerformanceFrequency(&v))
+            freq=v.QuadPart;
+        if (!freq)
+            return 0;
+    }
+    if (!QueryPerformanceCounter(&v))
+        return 0;
+    return (unsigned) ((v.QuadPart*U64C(1000000000))/freq); // This is unlikely to cleanly wrap
```

Comment:
question: Why are you casting to unsigned when the return type is unsigned __int64?